cmd/registry-replacer/main.go

```diff
@@ -600,6 +635,88 @@ func pruneReplacements(config *api.ReleaseBuildConfiguration, filter asDirective
 	return utilerrors.NewAggregate(errs)
 }
 
+// when pruning unused base images we'll check the fully-resolved config to make sure an image is not used directly in
+// the config, or within any of the tests.
+func pruneUnusedBaseImages(config *api.ReleaseBuildConfiguration, resolvedConfig *api.ReleaseBuildConfiguration) {
+	usedBaseImages := sets.NewString()
```

Comment:
Intuitively I would have thought that this won't need the resolved config, as steps shouldn't really reference images from base_images, am I wrong about that?

Reply:
Yep, they can and do - but it's not ideal. More often they refer to images from the release.